pkg/segment/structs/segstructs.go

```diff
@@ -1084,6 +1084,17 @@ func (qa *QueryAggregators) HasListFunc() bool {
 	return false
 }
 
+func HasNumTypeAggForColumn(measureOperations []*MeasureAggregator, column string) bool {
```

Comment:
nit: You've defined this function in the middle of a list of `QueryAggregators` methods. I'd suggest moving this definition somewhere else, or maybe it makes sense to make this a `QueryAggregators` method.

Reply:
Moved the definition out of `QueryAggregators `block